### ISSUE TITLE
TSLint preserves symlinks when searching for custom rules

### ISSUE THREAD

**Issue by nickpape-msft** [palantir/tslint#4294]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.4.2
- __Running TSLint via__: Node.js API

Related issues:
https://github.com/pnpm/pnpm/issues/1487
https://github.com/SharePoint/sp-dev-docs/issues/2934

#### TypeScript code being linted

You can find a repro here: https://github.com/nickpape-msft/pnpm-install-issue

Just run `pnpm install` and then `gulp`.

> **Note:** this issue occurs primarily when using different package managers that set up their node_modules differently (i.e. using symlinks). You can see https://github.com/pnpm/pnpm/issues/1487 for more details.

Basically, our project's `tslint.json` references `@microsoft/sp-tslint-rules`, which in turn references `tslint-microsoft-contrib` for custom rules. However, it seems tslint is preserving symlinks when trying to resolve that module (even though the default Node.js behavior is to resolve symlinks).

```ts
// code snippet
// n/a
```

with `tslint.json` configuration:

```json
{
  "$schema": "http://json.schemastore.org/tslint",

  "rulesDirectory": [
    "tslint-microsoft-contrib",
    "./lib/rules"
  ]
}
```

#### Actual behavior

TsLint throws an error because it cannot find the rules for `tslint-microsoft-contrib`.

#### Expected behavior

We are able to resolve the `tslint-microsoft-contrib` rules.

**Comment by iclanton** [comment]
This issue is fixed in https://github.com/palantir/tslint/pull/4295

**Comment by JoshuaKGoldberg** [comment]
Closing the issue as the PR has been merged in; #4340 tracks a new release to make this available.
